runtime/Go/antlr/tokenstream_rewriter_test.go

```diff
@@ -0,0 +1,392 @@
+			r.InsertBeforeDefault(1, "x")
+			r.ReplaceDefault(2, 3, "foo")
+		}),
+		NewLexerTest("abcc", "axbfoo", "LeaveAloneDisjointInsert2",
+		func(r *TokenStreamRewriter){
+			r.ReplaceDefault(2, 3, "foo")
+			r.InsertBeforeDefault(1, "x")
+		}),
+		NewLexerTest("abc", "aby", "InsertBeforeTokenThenDeleteThatToken",
+		func(r *TokenStreamRewriter){
+			r.InsertBeforeDefault(2, "y")
+			r.DeleteDefaultPos(2)
+		}),
+		NewLexerTest("aa", "<b>a</b><b>a</b>", "DistinguishBetweenInsertAfterAndInsertBeforeToPreserverOrder",
+		func(r *TokenStreamRewriter){
+			r.InsertBeforeDefault(0, "<b>")
+			r.InsertAfterDefault(0, "</b>")
+			r.InsertBeforeDefault(1, "<b>")
+			r.InsertAfterDefault(1,"</b>")
+		}),
+		NewLexerTest("aa", "<b><p>a</p></b><b>a</b>", "DistinguishBetweenInsertAfterAndInsertBeforeToPreserverOrder2",
+		func(r *TokenStreamRewriter){
+			r.InsertBeforeDefault(0, "<p>")
+			r.InsertBeforeDefault(0, "<b>")
+			r.InsertAfterDefault(0, "</p>")
+			r.InsertAfterDefault(0, "</b>")
+			r.InsertBeforeDefault(1, "<b>")
+			r.InsertAfterDefault(1,"</b>")
+		}),
+		NewLexerTest("ab", "<div><b><p>a</p></b></div>!b", "DistinguishBetweenInsertAfterAndInsertBeforeToPreserverOrder2",
+		func(r *TokenStreamRewriter){
+			r.InsertBeforeDefault(0, "<p>")
+			r.InsertBeforeDefault(0, "<b>")
+			r.InsertBeforeDefault(0, "<div>")
+			r.InsertAfterDefault(0, "</p>")
+			r.InsertAfterDefault(0, "</b>")
+			r.InsertAfterDefault(0, "</div>")
+			r.InsertBeforeDefault(1, "!")
+		}),
+	}
+
+
+	for _,c := range tests{
+		t.Run(c.description,func(t *testing.T) {
+			rewriter := prepare_rewriter(c.input)
+			c.ops(rewriter)
+			if len(c.expected_exception)>0{
+				panic_tester(t, c.expected_exception, rewriter)
+			}else{
+				result := rewriter.GetTextDefault()
+				if result!=c.expected{
+					t.Errorf("Expected:%s | Result: %s", c.expected, result)
+				}
+			}
+		} )
+	}
+}
+
+
+// Suppress unused import error
+var _ = fmt.Printf
+var _ = unicode.IsLetter
+
+var serializedLexerAtn = []uint16{
+	3, 24715, 42794, 33075, 47597, 16764, 15335, 30598, 22884, 2, 5, 15, 8,
+	1, 4, 2, 9, 2, 4, 3, 9, 3, 4, 4, 9, 4, 3, 2, 3, 2, 3, 3, 3, 3, 3, 4, 3,
+	4, 2, 2, 5, 3, 3, 5, 4, 7, 5, 3, 2, 2, 2, 14, 2, 3, 3, 2, 2, 2, 2, 5, 3,
+	2, 2, 2, 2, 7, 3, 2, 2, 2, 3, 9, 3, 2, 2, 2, 5, 11, 3, 2, 2, 2, 7, 13,
+	3, 2, 2, 2, 9, 10, 7, 99, 2, 2, 10, 4, 3, 2, 2, 2, 11, 12, 7, 100, 2, 2,
+	12, 6, 3, 2, 2, 2, 13, 14, 7, 101, 2, 2, 14, 8, 3, 2, 2, 2, 3, 2, 2,
+}
+
+var lexerDeserializer = NewATNDeserializer(nil)
+var lexerAtn = lexerDeserializer.DeserializeFromUInt16(serializedLexerAtn)
+
+var lexerChannelNames = []string{
+	"DEFAULT_TOKEN_CHANNEL", "HIDDEN",
+}
+
+var lexerModeNames = []string{
+	"DEFAULT_MODE",
+}
+
+var lexerLiteralNames = []string{
+	"", "'a'", "'b'", "'c'",
+}
+
+var lexerSymbolicNames = []string{
+	"", "A", "B", "C",
+}
+
+var lexerRuleNames = []string{
+	"A", "B", "C",
+}
+
+type LexerA struct {
+	*BaseLexer
+	channelNames []string
+	modeNames    []string
+	// TODO: EOF string
```

Comment:
This TODO (and the other) is not actionable (i.e. it's not clear what needs to be done to finish whatever needs to be done).